### PR TITLE
Updating spec with StreamInvocation, removing StreamCompletion

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: specs/HubProtocol.md [hunk 1]
```diff
@@ -72,30 +72,30 @@ Invocations can be marked as "Non-Blocking" in the `Invocation` message, which i
...
 
-If a Callee is going to stream results, it **MUST** send each individual result in a separate `StreamItem` message, and complete the invocation with a `StreamCompletion`. If the Callee is going to return a single result, it **MUST** not send any `StreamItem` messages, and **MUST** send the single result in a `Completion` message. This is to ensure that the Caller can unambiguously determine the intended streaming behavior of the method.
+If a Callee is going to stream results, it **MUST** send each individual result in a separate `StreamItem` message, and complete the invocation with a `Completion`. If the Callee is going to return a single result, it **MUST** not send any `StreamItem` messages, and **MUST** send the single result in a `Completion` message. If the Callee receives an `Invocation` message for a method that would yield multiple results or the Callee receives a `StreamInvocationMessage` for a method that would return single result it **MUST** complete the invocation with a `Completion` message containing an error.
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: "for a method that would return **a** single result it"

File: specs/HubProtocol.md [hunk 2]
```diff
@@ -275,6 +275,29 @@ Example (Non-Blocking):
...
+* `type` - A `Number` with the literal value 4, indicating that this message is a StreamInvocation.
+* `invocationId` - A `String` encoding the `Invocation ID` for a message.
+* `target` - A `String` encoding the `Target` name, as expected by the Callee's Binder
+* `arguments` - An `Array` containing arguments to apply to the method referred to in Target. This is a sequence of JSON `Token`s, encoded as indicated below in the "JSON Payload Encoding" section
+
+Example:
+
+```json
+{
+    "type": 1,
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: 4

File: specs/HubProtocol.md [hunk 3]
```diff
@@ -275,6 +275,29 @@ Example (Non-Blocking):
 }
 ```
 
+### StreamInvocation Message Encoding
+
+A `StreamInvocation` message is a JSON object with the following properties:
+
+* `type` - A `Number` with the literal value 4, indicating that this message is a StreamInvocation.
+* `invocationId` - A `String` encoding the `Invocation ID` for a message.
+* `target` - A `String` encoding the `Target` name, as expected by the Callee's Binder
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: period at the end

File: specs/HubProtocol.md [hunk 4]
```diff
@@ -275,6 +275,29 @@ Example (Non-Blocking):
...
+* `arguments` - An `Array` containing arguments to apply to the method referred to in Target. This is a sequence of JSON `Token`s, encoded as indicated below in the "JSON Payload Encoding" section
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: period at the end

File: specs/HubProtocol.md [hunk 5]
```diff
@@ -438,6 +434,46 @@ is decoded as follows:
...
 
+### StreamInvocation Message Encoding
+
+`StreamInvocation` messages have the following structure:
+
+```
+[4, InvocationId, Target, [Arguments]]
+```
+
+* `4` - Message Type - `4` indicates this is a `StreamInvocation` message
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: period at the end

File: specs/HubProtocol.md [hunk 6]
```diff
@@ -438,6 +434,46 @@ is decoded as follows:
...
+* InvocationId - A `String` encoding the Invocation ID for the message
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: period at the end

File: specs/HubProtocol.md [hunk 7]
```diff
@@ -438,6 +434,46 @@ is decoded as follows:
...
+* Target - A `String` encoding the Target name, as expected by the Callee's Binder
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: period at the end

File: specs/HubProtocol.md [hunk 8]
```diff
@@ -438,6 +434,46 @@ is decoded as follows:
...
+
+The following payload
+
+```
+0x95 0x04 0xa3 0x78 0x79 0x7a 0xa6 0x6d 0x65 0x74 0x68 0x6f 0x64 0x91 0x2a
+```
+
+is decoded as follows:
+
+* `0x94` - 5-element array
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: 0x95

File: specs/HubProtocol.md [hunk 9]
```diff
@@ -438,6 +434,46 @@ is decoded as follows:
...
+* `0x04` - `4` (Message Type - `StreamInvocation` message)
+* `0xa3` - string of length 3 (InvocationId)
+* `0x78` - `x`
+* `0x79` - `y`
+* `0x7a` - `z`
+* `0xc3` - `true` (NonBlocking)
```
Comment (BrennanConroy) [inline specs/HubProtocol.md]: remove?

File: specs/HubProtocol.md (hunk 8 above)
Comment (BrennanConroy) [reply specs/HubProtocol.md]: If you remove `NonBlocking` then should be 4-element array

**Comment by moozzyk** [comment]
Merged as 18f308fb05eabcc72ebc680a5c21c04b961a387e

**Pull request closed without merging**